MekHQ/unittests/mekhq/gui/model/UnitTableModelTest.java

```diff
@@ -0,0 +1,180 @@
+     }
+
+     /**
+      * Returns a list of people with the specified person count.
+      *
+      * @param personCount the number of persons to include in the list
+      * @return a list of people with the specified person count
+      */
+     private List<Person> getPeople(int personCount) {
+         return Collections.nCopies(personCount, person);
+     }
+
+     /**
+      * Initializes the gunners of a mock unit.
+      *
+      * @param gunnerNeeds The total number of gunner needs for the unit.
+      * @param gunnersAssigned The number of gunners currently assigned to the unit.
+      */
+     private void initializeGunners(int gunnerNeeds, int gunnersAssigned) {
+         when(unit.getTotalGunnerNeeds()).thenReturn(gunnerNeeds);
+         when(unit.getGunners()).thenReturn(getPeople(gunnersAssigned));
+     }
+
+     /**
+      * Initializes the crew of a mock unit.
+      *
+      * @param crewNeeds The total number of crew members needed for the unit.
+      * @param crewAssigned The number of crew members already assigned to the unit.
+      */
+     private void initializeCrew(int crewNeeds, int crewAssigned) {
+         when(unit.getTotalCrewNeeds()).thenReturn(crewNeeds);
+         when(unit.getCrew()).thenReturn(getPeople(crewAssigned));
+     }
+
+     /**
+      * Initializes the entity class based on the given parameters.
+      *
+      * @param isSpaceStation {@code true} if the entity is a space station, {@code false} if it is
+      *                       a JumpShip
+      * @param navigator      optional parameter indicating the navigator of the entity, can be
+      *                       {@code null}
+      */
+     private void initializeEntityClass(boolean isSpaceStation, @Nullable Person navigator) {
+         if (isSpaceStation) {
+             when(unit.getEntity()).thenReturn(mock(SpaceStation.class));
+         } else {
+             when(unit.getEntity()).thenReturn(mock(Jumpship.class));
+         }
+
+         when(unit.getNavigator()).thenReturn(navigator);
+     }
+
+     // In the following test the appendReport() method is called, and its response is checked
+     // against expected behavior
+
+     @Test
+     public void appendReportTest() {
+         StringBuilder report = new StringBuilder("<html>");
+         String title = "roleName";
+         int assigned = 5;
+         int needed = 10;
+
+         appendReport(report, title, assigned, needed);
+
+         String expected = "<html><br><b>roleName: </b>5/10";
+         assertEquals(expected, report.toString());
+     }
+
+     // In the following tests the getCrewTooltip() method is called, and its response is checked
+     // against expected behavior
+
+     @Test
+     public void noGunner() {
+         initializeGunners(0,0);
+         initializeEntityClass(true, null);
+         initializeCrew(0,0);
+
+         String expected = "<html><br><b>Drivers: </b>1/1</html>";
```

Comment:
I don't do much in the way of unit tests, so please take it with a grain of salt: These tests check GUI output. It seems much more likely that they'll need an update for someone improving the tooltip than they would find an error.

Reply:
Any suggestions on how I could improve them? I only learned how to do Unit Tests over the weekend, so I'd greatly appreciate any suggestions :)